tests/test_io.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+import pytest
+from packaging.version import Version
+
+import spherely
+
+
+def test_from_wkt():
+    result = spherely.from_wkt(["POINT (1 1)", "POINT(2 2)", "POINT(3 3)"])
+    expected = spherely.create([1, 2, 3], [1, 2, 3])
+    # object equality does not yet work
+    # np.testing.assert_array_equal(result, expected)
+    assert spherely.equals(result, expected).all()
+
+    # from explicit object dtype
+    result = spherely.from_wkt(
+        np.array(["POINT (1 1)", "POINT(2 2)", "POINT(3 3)"], dtype=object)
+    )
+    assert spherely.equals(result, expected).all()
+
+    # from numpy string dtype
+    result = spherely.from_wkt(
+        np.array(["POINT (1 1)", "POINT(2 2)", "POINT(3 3)"], dtype="U")
+    )
+    assert spherely.equals(result, expected).all()
+
+
+def test_from_wkt_invalid():
+    # TODO can we provide better error type?
+    with pytest.raises(RuntimeError):
+        spherely.from_wkt(["POINT (1)"])
+
+
+def test_from_wkt_wrong_type():
+    with pytest.raises(TypeError, match="expected bytes, int found"):
+        spherely.from_wkt([1])
+
+    # TODO support missing values
+    with pytest.raises(TypeError, match="expected bytes, NoneType found"):
+        spherely.from_wkt(["POINT (1 1)", None])
+
+
+polygon_with_bad_hole_wkt = (
+    "POLYGON "
+    "((20 35, 10 30, 10 10, 30 5, 45 20, 20 35),"
+    "(30 20, 20 25, 20 15, 30 20))"
+)
+
+
+@pytest.mark.skipif(
+    Version(spherely.__s2geography_version__) < Version("0.2.0"),
+    reason="Needs s2geography >= 0.2.0",
+)
+def test_from_wkt_oriented():
+    # by default re-orients the inner ring
+    result = spherely.from_wkt(polygon_with_bad_hole_wkt)
+    assert (
+        str(result)
+        == "POLYGON ((20 35, 10 30, 10 10, 30 5, 45 20, 20 35), (20 15, 20 25, 30 20, 20 15))"
+    )
+
+    # if we force to not orient, we get an error
+    with pytest.raises(RuntimeError, match="Inconsistent loop orientations detected"):
+        spherely.from_wkt(polygon_with_bad_hole_wkt, oriented=True)
```

Comment:
Out of curiosity, is there a great performance gain when setting `oriented=True` and when we know that all ring are consistently oriented? I haven't checked the implementation in s2geography (and not much in s2geometry), but I assume that the orientations are checked anyway if such case may trigger raising an error.

Maybe are there other reasons of setting `oriented=True`? I.e., when do we want to avoid implicit re-ordering of the rings?

Reply:
We might also later uniformize across the library the type of error raised (in #51 a ValueError is raised when a polygon has invalid edges).

Reply:
I don't know (without trying out) if there is a performance difference, but I think that the `oriented=True` does allow you to create different geographies? (if you have one that is larger than half of the globe, the default constructor would always normalize it to represent the smaller polygon)

And I can also imagine that just checking that the orientation of all loops are the same is still cheaper than a full normalization (but that is indeed something to check)

Reply:
> We might also later uniformize across the library the type of error raised (in #51 a ValueError is raised when a polygon has invalid edges).

Yeah, in this case the error comes from s2geography (in theory I could catch the error and re-throw it with a different class, because those RuntimeErrors are a bit strange)